runtime/index.d.ts

```diff
@@ -35,3 +36,22 @@ declare module "@sapper/service-worker" {
 	export const shell: string[];
 	export const routes: Array<{ pattern: RegExp }>;
 }
+
+declare module "@sapper/common" {
+	export interface PreloadContext {
+		fetch: (url: string, options?: any) => Promise<any>;
```

Comment:
Would it be appropriate to use the `node-fetch` types here instead of typing opts and res as any?

```ts
    fetch: (url: RequestInfo, init?: RequestInit) => Promise<Response>;
```

Reply:
That seems like it would probably work for `Response`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/07b2759140ad390e14801cfcc5ed9ff94182e7bb/types/node-fetch/index.d.ts#L167

For the options, `node-fetch` extends the whatwg/fetch standard options with its own:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/07b2759140ad390e14801cfcc5ed9ff94182e7bb/types/node-fetch/index.d.ts#L46

That would only work on the server and I'm not sure we'd want to expose the underlying implementation to that degree. Perhaps we should just define our own interface for that? 
```
export interface FetchOpts {
    body?: BodyInit;
    headers?: HeadersInit;
    method?: string;
    redirect?: RequestRedirect;
    signal?: AbortSignal | null;
}
```

Or maybe we can find types for the browser `fetch` that we can use instead?

Reply:
Ah, I didn't realise node-fetch was a superset of the fetch standard. In that case it's safer to use the equivalent signature for native fetch

https://microsoft.github.io/PowerBI-JavaScript/modules/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.html#fetch

Reply:
… in which case I think `fetch: WindowOrWorkerGlobalScope["fetch"]` would work

Reply:
Just wondering — why wasn’t this typed in the end?

Reply:
As mentioned below I felt there was no concensus yet on what the typings should be. Feel free to open another PR for a dedicated discussion around this and possibly changing it.

Reply:
would `WindowOrWorkerGlobalScope` work on the server-side? I'm not that familiar with typescript, but it sounds like a browser thing